src/OpenSage.Game/Logic/Object/Behaviors/FireWeaponWhenDeadBehavior.cs

```diff
@@ -1,68 +1,97 @@
+                WeaponSlot.Primary,
+                _gameObject.GameContext);
 
-            reader.PersistObject(_upgradeLogic);
+            deathWeapon.Fire();
         }
     }
 
-    public sealed class FireWeaponWhenDeadBehaviorModuleData : UpgradeModuleData
+    internal override void Load(StatePersister reader)
     {
-        internal static FireWeaponWhenDeadBehaviorModuleData Parse(IniParser parser) => parser.ParseBlock(FieldParseTable);
-
-        private static new readonly IniParseTable<FireWeaponWhenDeadBehaviorModuleData> FieldParseTable = UpgradeModuleData.FieldParseTable
-            .Concat(new IniParseTable<FireWeaponWhenDeadBehaviorModuleData>
-            {
-                { "RequiredStatus", (parser, x) => x.RequiredStatus = parser.ParseEnum<ObjectStatus>() },
-                { "ExemptStatus", (parser, x) => x.ExemptStatus = parser.ParseEnum<ObjectStatus>() },
-                { "DeathWeapon", (parser, x) => x.DeathWeapon = parser.ParseAssetReference() },
-                { "DeathTypes", (parser, x) => x.DeathTypes = parser.ParseEnumBitArray<DeathType>() },
-                { "DelayTime", (parser, x) => x.DelayTime = parser.ParseInteger() },
-                { "WeaponOffset", (parser, x) => x.WeaponOffset = parser.ParseVector3() },
-            });
-
-        public ObjectStatus RequiredStatus { get; private set; }
-        public ObjectStatus ExemptStatus { get; private set; }
-        public string DeathWeapon { get; private set; }
-        public BitArray<DeathType> DeathTypes { get; private set; }
-
-        [AddedIn(SageGame.Bfme)]
-        public int DelayTime { get; private set; }
-
-        [AddedIn(SageGame.Bfme)]
-        public Vector3 WeaponOffset { get; private set; }
-
-        internal override BehaviorModule CreateModule(GameObject gameObject, GameContext context)
+        reader.PersistVersion(1);
+
+        reader.BeginObject("Base");
+        base.Load(reader);
+        reader.EndObject();
+
+        reader.PersistObject(UpgradeLogic);
+    }
+}
+
+public sealed class FireWeaponWhenDeadBehaviorModuleData : UpgradeModuleData
+{
+    internal static FireWeaponWhenDeadBehaviorModuleData Parse(IniParser parser) => parser.ParseBlock(FieldParseTable);
+
+    private static new readonly IniParseTable<FireWeaponWhenDeadBehaviorModuleData> FieldParseTable = UpgradeModuleData.FieldParseTable
+        .Concat(new IniParseTableChild<FireWeaponWhenDeadBehaviorModuleData, DieLogicData>(x => x.DieData, DieLogicData.FieldParseTable))
+        .Concat(new IniParseTable<FireWeaponWhenDeadBehaviorModuleData>
         {
-            return new FireWeaponWhenDeadBehavior(this);
-        }
+            { "DeathWeapon", (parser, x) => x.DeathWeapon = parser.ParseWeaponTemplateReference() },
+            { "DelayTime", (parser, x) => x.DelayTime = parser.ParseInteger() },
+            { "WeaponOffset", (parser, x) => x.WeaponOffset = parser.ParseVector3() },
+        });
+
+    public DieLogicData DieData { get; } = new();
```

Comment:
Is it possible to just push this all the way down to the base class? An easy way to test would be to edit the inis and apply one of the properties to some other DieModule type which doesn't have them documented and see if they work. I noticed with the CrateBehavior stuff I was working on that a lot of the data actually worked in a base class and so the properties and logic could be pushed down.

I guess even if the original game doesn't accommodate them, there's no reason we couldn't. Might clean up the code too!